### ISSUE TITLE
Possible Bug in NDJWriter

### ISSUE THREAD

**Issue by bwriordan** [EducationalTestingService/skll#416]
When creating FeatureSet instances using columns and index values from a data, `astype('str')` seem to be necessary or else SKLL will throw an error during NDJWriter.write().

Test code:

```python
dftest = pd.DataFrame({"id":["1", "2"], "score":[1,2], "text":["a b", "b c"]})
dftest.set_index("id", inplace=True)
test_feat_dict_list = [{'a':1.0, 'b':1.0}, {'b':1.0, 'c':1.0}]
test_dict_vectorizer = DictVectorizer()
Xtest = test_dict_vectorizer.fit_transform(test_feat_dict_list)
fs_test = FeatureSet('test', ids=dftest.index.values, labels=dftest['score'].values, features=Xtest, vectorizer=test_dict_vectorizer)
test_writer = NDJWriter('skll_test.jsonlines', fs_test)
test_writer.write()
```

Produces the following error:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/briordan/anaconda/envs/nwd/lib/python3.5/site-packages/skll/data/writers.py", line 133, in write
    self._write_subset(self.path, None)
  File "/Users/briordan/anaconda/envs/nwd/lib/python3.5/site-packages/skll/data/writers.py", line 175, in _write_subset
    self._write_line(id_, label_, feat_dict, output_file)
  File "/Users/briordan/anaconda/envs/nwd/lib/python3.5/site-packages/skll/data/writers.py", line 590, in _write_line
    example_dict['id'] = np.asscalar(id_)
  File "/Users/briordan/anaconda/envs/nwd/lib/python3.5/site-packages/numpy/lib/type_check.py", line 482, in asscalar
    return a.item()
AttributeError: 'str' object has no attribute 'item'
```

**Comment by desilinguist** [comment]
This is most likely because pandas uses the `object` type for strings and that may just be incompatible with SKLL. Perhaps we can built-in a workaround or just update the documentation to say that if you are passing in `pandas` values, make sure to convert them.

**Comment by desilinguist** [comment]
Keep it open please.

**Comment by desilinguist** [comment]
Still keep it open.

**Comment by desilinguist** [comment]
Keep it open.

**Comment by desilinguist** [comment]
Okay, I looked more into this and turns out the main issue is that we use `asscalar()` in the `_write_line()` method of `NDJWriter` since we are almost always expecting a numpy datatype there and so we need to convert that to a scalar otherwise `json.dumps()` will complain. However, in the `pandas` case, we already get a scalar so I just need to add a bit of `try ... except` to handle that edge case. PR forthcoming.

**Comment by desilinguist** [comment]
Addressed by #440.
